### ISSUE TITLE
bug: __mocks__ copied to output target

### ISSUE THREAD

**Issue by mrtnmgs** [ionic-team/stencil#5781]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.18.0

### Current Behavior

Stencil copies over __mocks__ directories to `dist` and throws warnings about duplicate mocks.
The mocks are `./src/global/services/__mocks__` and  `./src/global/__mocks__`
Stencil's console warning:
```
jest-haste-map: duplicate manual mock found: functions
  The following files share their name; please delete one of them:
    * <rootDir>\src\global\__mocks__\functions.tsx
    * <rootDir>\dist\collection\global\__mocks__\functions.js
    * 
jest-haste-map: duplicate manual mock found: http-service
  The following files share their name; please delete one of them:
    * <rootDir>\src\global\services\__mocks__\http-service.tsx
    * <rootDir>\dist\collection\global\services\__mocks__\http-service.js
```

I wondered if I wasn't supposed to write the mocks in TypeScript, but same thing with .js files:

```
jest-haste-map: duplicate manual mock found: functions
  The following files share their name; please delete one of them:
    * <rootDir>\src\global\__mocks__\functions.js
    * <rootDir>\dist\collection\global\__mocks__\functions.js

jest-haste-map: duplicate manual mock found: http-service
  The following files share their name; please delete one of them:
    * <rootDir>\src\global\services\__mocks__\http-service.js
    * <rootDir>\dist\collection\global\services\__mocks__\http-service.js
```

I couldn't find a config option that would allow to explicitly exclude these directories.

### Expected Behavior

`__mocks__` directories should  be excluded when copying files to build directories

### System Info

```shell
Windows, npm 9.7.2
```

### Steps to Reproduce

create file `src/global/functions.tsx` and matching mock `src/global/__mocks__/functions.tsx`. run `npm test`.

### Code Reproduction URL

https://github.com/mrtnmgs/mockbug

**Comment by alicewriteswrongs** [comment]
Hey @mrtnmgs thanks for reporting this, I was just able to confirm that the files in the `__mocks__` directory are being copied over. This isn't specific behavior to that directory though - any code in `src/` will be copied to the dist output target by Stencil actually. So I think in order to change this behavior we'd need to add an `ignore` option or something.

I'm going to label this a feature request so we can gauge community interest in such a feature!

**Comment by mrtnmgs** [comment]
Thanks @alicewriteswrongs. That feature sounds nice, but right now things don't work properly by default. There's no reason to copy `__mocks__` directories over, and it creates an issue. It doesn't sound right to me to ask the user to configure something in order to fix it... Shouldn't `__mocks__` directories be ignored by default?

**Comment by christian-bromann** [comment]
A fix for this bug has been published in Stencil [`v4.21.0`](https://github.com/ionic-team/stencil/releases/tag/v4.21.0)!
